index.js

```diff
@@ -5,39 +5,36 @@ const timer = document.querySelector('#timer')
 let counter = 0;
 let memoryCounter = 5;
 
-// Timing the counters
-const decreaseTimer = setInterval(function(){
-    if(memoryCounter>0){
+// For modifying cards
+const makeCardsClickable = setTimeout(function(){
+    cards.forEach(card=>{
+        card.addEventListener('click',flipCard); // Makes cards clickable after the timer expires
+        card.children[1].style.display="block"; // Hides them after the timer expires
+    })
+},memoryCounter*1000);
+
+// For timing the counters
+const interval = setInterval(function(){
+    if(memoryCounter>0){ // Makes a check on memoryCounter and stops after 5 secs
         memoryCounter--;
         viewAllCards();
```

Comment:
this `viewAllCards` function is called every time until the memory counter is zero, which is not good for performance. Add some conditions so that it do not run unnecessarily